### ISSUE TITLE
Crashes Electron 8.2.x

### ISSUE THREAD

**Issue by Slapbox** [streetsidesoftware/regexp-worker#13]
No error, just leads to blank page and disconnected devtools.

**Crashed code (from readme):**

```js
  const defaultTimeOutMs = 1000;
  const worker = await new RegExpWorker(defaultTimeOutMs);

  // Find all words in some text
  const words = await worker.execRegExp(/\b\w+/g, 'Lots of text ...');
```

Node worker threads more broadly work without issue. Cool project, hope to be able to use it in the future.

**Comment by Jason3S** [comment]
I'm happy you tried it out! 
This cannot be used directly in Electron. It needs to be run in a separate process.
It needs to be able to Kill the thread if the regexp gets stuck.
I have been successfully using it with Electron and https://github.com/microsoft/vscode-languageserver-node

**Comment by Slapbox** [comment]
Hey @Jason3S!

I'm confused about why this wouldn't be compatible with Electron, given that Electron can spawn Node worker_threads, which is what this library seems to depend on?

I went in a different direction in the end, but I'm curious why it wouldn't be compatible?

**Comment by Jason3S** [comment]
From this doc: https://www.electronjs.org/docs/tutorial/multithreading it seems that Electron works with web_workers instead of node worker_threads. When I tried it directly within Electron, I wasn't able to get a work_thread to start.

I saw this: https://github.com/electron/electron/issues/17030 and https://github.com/electron/electron/issues/18540

**Comment by Slapbox** [comment]
Hey @Jason3S thanks for the follow-up and sorry to take literally months to get back to you. Electron actually does support the native node workers now - since whatever version uses Node 12. We are currently using node workers in our app. 

I wish I could provide some more useful data on the error we encountered, but unfortunately there were no errors - just the process crash. Thanks again for sharing your work!

**Comment by Jason3S** [comment]
@Slapbox,

Out of curiosity, did you happen to WebPack or bundle it in any way?

**Comment by Slapbox** [comment]
@Jason3S  yes it was imported (or required) within a Webpack build system.

**Comment by Jason3S** [comment]
Thank you. That explains the crash. worker_theads must be in their own file. That is how NodeJS does it.

When this package is bundled in a Webpack, the filename to run becomes the app, effectively restarting the full electron app every time a worker thread is started. As expected, this crashes.

**Comment by Jason3S** [comment]
This is now fixed in 4.x. It is possible to bundle the package for node or the browser.
